pkg/cluster-runtime/apps.go

```diff
@@ -98,38 +96,39 @@ func (i AppInstaller) LaunchClusterImage(master0 net.IP, launchCmds []string) er
 			return fmt.Errorf("failed to render launch cmd: %v", err)
 		}
 
-		if err = i.infraDriver.CmdAsync(master0, fmt.Sprintf(common.CdAndExecCmd, clusterRootfs, cmdline)); err != nil {
+		if err = i.infraDriver.CmdAsync(master0, fmt.Sprintf(common.CdAndExecCmd, appPath, cmdline)); err != nil {
 			return err
 		}
 	}
 
-	return i.save()
+	return i.save(common.GetDefaultApplicationFile())
 }
 
 // todo save image info to disk or api server, we need new interface to do this.
-func (i AppInstaller) save() error {
-	var extensionList []v12.ImageExtension
-	applicationFile := common.GetDefaultApplicationFile()
+func (i AppInstaller) save(applicationFile string) error {
+	f, err := os.OpenFile(applicationFile, os.O_RDWR|os.O_APPEND|os.O_CREATE, 0600)
+	defer func() {
```

Comment:
use `defer f.Close() ` is OK

Reply:
golang lint will check the func return error ,to make sure each error was handled . so i use defer func()